contracts/extensions/Auctions/DutchAuction.sol

```diff
@@ -0,0 +1,483 @@
+    bytes32 auctionId = keccak256(abi.encodePacked(collection, item));
+    DutchAuctionData memory auctionDetails = auctions[auctionId];
+
+    if (auctionDetails.info == 0) {
+      revert INVALID_AUCTION();
+    }
+
+    uint256 expiration = uint256(uint64(auctionDetails.prices >> 192));
+
+    if (block.timestamp > deploymentOffset + expiration) {
+      revert AUCTION_ENDED();
+    }
+
+    if (auctionDetails.bid != 0) {
+      uint256 currentBidAmount = uint96(auctionDetails.bid >> 160);
+      if (currentBidAmount >= msg.value) {
+        revert INVALID_BID();
+      }
+
+      payable(address(uint160(auctionDetails.bid))).transfer(currentBidAmount);
+    } else {
+      uint256 endingPrice = uint256(uint96(auctionDetails.prices >> 96));
+      // TODO: consider allowing this as a means of caputuring market info
+      if (endingPrice > msg.value) {
+        revert INVALID_BID();
+      }
+    }
+
+    uint256 newBid = uint256(uint160(msg.sender));
+    newBid |= uint256(uint96(msg.value)) << 160;
+
+    auctions[auctionId].bid = newBid;
+
+    // TODO: consider allowing a bid over the current minimum price to tranfer the token immediately without calling settle
+
+    emit PlaceBid(msg.sender, collection, item, msg.value, _memo);
+  }
+
+  /**
+   * @notice Settles the auction after expiration if no valid bids were received by sending the item back to the seller. If a valid bid matches the current price at the time of settle call, the item is sent to the bidder. Proceeds will be distributed separately by calling `distributeProceeds`.
+   *
+   * @param collection ERC721 contract.
+   * @param item Token id to settle.
+   */
+  function settle(
+    IERC721 collection,
+    uint256 item,
+    string calldata _memo
+  ) external override nonReentrant {
+    bytes32 auctionId = keccak256(abi.encodePacked(collection, item));
+    DutchAuctionData memory auctionDetails = auctions[auctionId];
+
+    if (auctionDetails.info == 0) {
+      revert INVALID_AUCTION();
+    }
+
+    uint256 lastBidAmount = uint256(uint96(auctionDetails.bid >> 160));
+    uint256 minSettlePrice = currentPrice(collection, item);
+
+    if (minSettlePrice < lastBidAmount) {
+      address buyer = address(uint160(auctionDetails.bid));
+
+      collection.transferFrom(address(this), buyer, item);
+
+      emit ConcludeAuction(
+        address(uint160(auctionDetails.info)),
+        buyer,
+        collection,
+        item,
+        lastBidAmount,
+        _memo
+      );
+    } else {
+      uint256 expiration = uint256(uint64(auctionDetails.prices >> 192));
+      if (block.timestamp > deploymentOffset + expiration) {
+        // NOTE: return token back to seller after auction expiration if highest bid was below settlement price
+        collection.transferFrom(address(this), address(uint160(auctionDetails.info)), item);
+
+        delete auctions[auctionId];
+        delete auctionSplits[auctionId];
+
+        emit ConcludeAuction(
```

Comment:
nit: you can move the emit to below the if/else, that way you don't need to have 2 seperate ones

Reply:
they are different, one concludes with a buyer, the other with address(0).